dali/operators/reader/numpy_reader_op.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "dali/kernels/common/transpose.h"
+#include "dali/core/static_switch.h"
+#include "dali/operators/reader/numpy_reader_op.h"
+
+namespace dali {
+
+#define NUMPY_ALLOWED_TYPES \
+  (bool, uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float, float16, \
+  double)
+
+void NumpyReader::TransposeHelper(Tensor<CPUBackend>& output, const Tensor<CPUBackend>& input) {
+  auto& in_shape = input.shape();
+  auto& type = input.type();
+  uint16_t n_dims = in_shape.size();
+  std::vector<int> perm(n_dims);
+  std::vector<int64_t> out_shape(n_dims);
+  for (uint16_t i = 0; i < n_dims; ++i) {
```

Comment:
...likewise.

Reply:
It would warn signed-unsigned comparison.

Reply:
fixed